apps/src/code-studio/scriptAnnouncementsRedux.js

```diff
@@ -2,19 +2,28 @@ import { PropTypes } from 'react';
 import { NotificationType } from '@cdo/apps/templates/Notification';
 
 const ADD_ANNOUNCEMENT = 'scriptAnnouncements/ADD_ANNOUNCEMENT';
-export const addAnnouncement = (notice, details, link, type) => ({
+
+export const addAnnouncement = (notice, details, link, type, visibility) => ({
   type: ADD_ANNOUNCEMENT,
   notice,
   details,
   link,
   announcementType: type,
+  visibilityType: visibility,
 });
 
+export const VisibilityType = {
+  teacher: 'Teacher-only',
+  student: 'Student-only',
+  teacherAndStudent: 'Teacher and student',
```

Comment:
I'm going to suggest that you omit dashes and spaces from these values and use casing and underscores instead.  The simplest thing would be to have the values equal the keys:
```
  teacher: 'teacher',
  student: 'student',
  teacherAndStudent: 'teacherAndStudent,
```
It doesn't look like there is a problem with the way you did things right now, but spaces, dashes and other characters can sometimes cause problems depending on what you pass these values into. So this is somewhat of an opinionated best practice I'm recommending, not a hard-and-fast rule.

Reply:
Talked to Dave we are going to leave it for now since its not causing issues as is currently.